### PR TITLE
Added Jenkinsfile

### PR DESCRIPTION
**Pull request by AdamSaleh**
In the end I didn't port over the integration tests (one case fails on wendy, and we don't run them on bob anyway, referenced in [RHMAP-18033](https://issues.jboss.org/browse/RHMAP-18033) )

### CODE REVIEW

File: test/setup_mongo_node.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+var DB = require('mongodb').Db;
```
Comment (grdryn) [inline test/setup_mongo_node.js]: This file doesn't look to get called by anything, does it? If it's intended to be called by mongo as a setup step before running the tests, consider wrapping it in a function and exporting that as `before`, and stop exiting the process at the end.

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,29 @@
+#!groovy
+
+// https://github.com/feedhenry/fh-pipeline-library
+@Library('fh-pipeline-library') _
+
+fhBuildNode() {
```
Comment (grdryn) [inline Jenkinsfile]: Could you change this `fhBuildNode() {` to:

```groovy
node('nodejs4') {

    step([$class: 'WsCleanup'])

    stage ('Checkout') {
        checkout scm
    }
```

File: Jenkinsfile [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    stage('Install Dependencies') {
+        npmInstall {}
+    }
+
+    stage('Unit Tests') {
+            sh 'grunt fh:unit'
+    }
+
+    stage('Integration Tests') {
+        print "Integration tests are disabled due to https://issues.jboss.org/browse/RHMAP-18033"
```
Comment (grdryn) [inline Jenkinsfile]: This is not a private internal project, so the Jira link should be publicly accessible in the FH project. Could you move that please, and make any necessary adjustments?

File: test/setup_mongo_node.js (hunk 1 above)
Comment (AdamSaleh) [reply test/setup_mongo_node.js]: It seemed to be necessary, but then I removed the setup clause, and forgot to remove the file.

**Review by grdryn — APPROVED** [review]
@AdamSaleh looks good to me! 👍 

The next steps are:
- merge
- disable the following jobs on bob and edit the description to say something like "Disabled on 2017-10-03 by asaleh, as the job on Wendy is now the required job for this repo."
  -  fh-mbaas-middleware_pullrequests
  - fh-mbaas-middleware_sonar-pullrequests
  - fh-mbaas-middleware_sonarqube-nightly-master
- Update this to remove the block from the jenkins-job-builder config (or I can do that if you prefer) https://github.com/fheng/jenkins-bob-builder/pull/343
- Update the settings in this repo to mark the new "pr-head" job as the required job, and remove the "default" job from being required
- Update the settings for this repo to remove the hooks for bob

**Pull request merged**